src/AirSpeedSensor.cc

```diff
@@ -0,0 +1,209 @@
+}
+
+//////////////////////////////////////////////////
+bool AirSpeedSensor::Load(const sdf::Sensor &_sdf)
+{
+  if (!Sensor::Load(_sdf))
+    return false;
+
+  if (_sdf.Type() != sdf::SensorType::AIR_SPEED)
+  {
+    gzerr << "Attempting to a load an AirSpeed sensor, but received "
+      << "a " << _sdf.TypeStr() << std::endl;
+    return false;
+  }
+
+  if (_sdf.AirSpeedSensor() == nullptr)
+  {
+    gzerr << "Attempting to a load an AirSpeed sensor, but received "
+      << "a null sensor." << std::endl;
+    return false;
+  }
+
+  if (this->Topic().empty())
+    this->SetTopic("/air_speed");
+
+  this->dataPtr->pub =
+      this->dataPtr->node.Advertise<msgs::AirSpeedSensor>(
+      this->Topic());
+
+  if (!this->dataPtr->pub)
+  {
+    gzerr << "Unable to create publisher on topic[" << this->Topic() << "].\n";
+    return false;
+  }
+
+  gzdbg << "Air speed for [" << this->Name() << "] advertised on ["
+         << this->Topic() << "]" << std::endl;
+
+  // Load the noise parameters
+  if (_sdf.AirSpeedSensor()->PressureNoise().Type() != sdf::NoiseType::NONE)
+  {
+    this->dataPtr->noises[AIR_SPEED_NOISE_PASCALS] =
+      NoiseFactory::NewNoiseModel(_sdf.AirSpeedSensor()->PressureNoise());
+  }
+
+  this->dataPtr->initialized = true;
+  return true;
+}
+
+//////////////////////////////////////////////////
+bool AirSpeedSensor::Load(sdf::ElementPtr _sdf)
+{
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(_sdf);
+  return this->Load(sdfSensor);
+}
+
+//////////////////////////////////////////////////
+bool AirSpeedSensor::Update(
+  const std::chrono::steady_clock::duration &_now)
+{
+  GZ_PROFILE("AirSpeedSensor::Update");
+  if (!this->dataPtr->initialized)
+  {
+    gzerr << "Not initialized, update ignored.\n";
+    return false;
+  }
+
+  msgs::AirSpeedSensor msg;
+  *msg.mutable_header()->mutable_stamp() = msgs::Convert(_now);
+  auto frame = msg.mutable_header()->add_data();
+  frame->set_key("frame_id");
+  frame->add_value(this->FrameId());
+
+  const float alt_rel = static_cast<float>(this->Pose().Pos().Z()); // Z-component from ENU
+  const float alt_amsl = (float)DEFAULT_HOME_ALT_AMSL + alt_rel;
+  const float temperature_local = TEMPERATURE_MSL - LAPSE_RATE * alt_amsl;
+  const float density_ratio = powf(TEMPERATURE_MSL / temperature_local , 4.256f);
+  const float air_density = AIR_DENSITY_MSL / density_ratio;
+
+	gz::math::Vector3d wind_vel_{0, 0, 0};
+  gz::math::Quaterniond veh_q_world_to_body = this->Pose().Rot();
+	gz::math::Vector3d air_vel_in_body_ = this->dataPtr->vel - veh_q_world_to_body.RotateVectorReverse(wind_vel_);
+	double diff_pressure = gz::math::sgn(air_vel_in_body_.X()) * 0.005f * air_density  * air_vel_in_body_.X() * air_vel_in_body_.X();
+
+  // Apply pressure noise
+  if (this->dataPtr->noises.find(AIR_SPEED_NOISE_PASCALS) !=
+      this->dataPtr->noises.end())
+  {
+    diff_pressure =
+      this->dataPtr->noises[AIR_SPEED_NOISE_PASCALS]->Apply(
+          diff_pressure);
+  }
+
```

Comment:
`AirSpeedSensorPrivate`'s `pressure` variable isn't being used anywhere. Should it be set here or should we remove the variable entirely?

```suggestion
  this->dataPtr->pressure = diff_pressure;
```

Reply:
Are getter/setters needed for pressure?